src/ol/format/WFS.js

```diff
@@ -938,7 +939,20 @@ function writeProperty(node, pair, objectStack) {
         GML32.prototype.writeGeometryElement(value, pair.value, objectStack);
       }
     } else {
-      writeStringTextNode(value, pair.value);
+      if (typeof pair.value === 'string') {
+        // Generate CDATA section to preserve whitespaces.
+        pair.value.split(']]>').forEach((part, i, a) => {
+          if (i < a.length - 1) {
+            part += ']]';
+          }
+          if (i > 0) {
+            part = '>' + part;
+          }
+          writeCDATASection(value, part);
```

Comment:
This won't work as expected. It will convert `"foo>]]bar"` to `<![CDATA[foo]]]]><![CDATA[>bar]]>` instead of `<![CDATA[foo>]]><![CDATA[]]bar]]>`.

I think this addition could benefit from a test case.